### ISSUE TITLE
Unexpected attribute error for terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks

### ISSUE THREAD

**Issue by jimmyzzxhlh** [hashicorp/vscode-terraform#1800]
### Extension Version

v2.32.0

### VS Code Version

Version: 1.91.1 (Universal)
Commit: f1e16e1e6214d7c44d078b1f0607b2388f29d729
Date: 2024-07-09T22:07:54.982Z
Electron: 29.4.0
ElectronBuildId: 9728852
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Darwin arm64 23.5.0

### Operating System

macOS 14.5 (23F79)

### Terraform Version

Terraform v1.7.1 on darwin_arm64

### Steps to Reproduce

The module `terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks` is starting to show red lines with all attributes as "Unexpected attributes". This was working before in the past couple of weeks. 

I tried to upgrade the module from 5.39.0 to 5.41.0 and the issue still exists.

![image](https://github.com/user-attachments/assets/b5ee6824-ec91-459d-ba41-2f59545eea7c)

### Expected Behavior

Shouldn't show any errors. 

### Actual Behavior

See aove.

### Terraform Configuration

```terraform
module "broker_service_account_role" {
  source                         = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
  role_name                      = "broker-service-account-role"
  attach_external_secrets_policy = true
  external_secrets_secrets_manager_arns = [
    "arn:aws:secretsmanager:us-west-2:637423350792:secret:production/*",
  ]
  # This is the KMS key ARN for aws/secretsmanager in the production account.
  external_secrets_kms_key_arns                      = ["arn:aws:kms:us-west-2:637423350792:key/..."]
  external_secrets_secrets_manager_create_permission = false

  oidc_providers = {
    ex = {
      provider_arn = module.broker_eks_production.oidc_provider_arn
      namespace_service_accounts = [
        "authbroker:authbroker-service-account",
      ]
    }
  }
}
```

### Help Wanted

- [ ] I'm interested in contributing a fix myself

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by jpalcisodeltek** [comment]
The double-slash in the module source address is being treated as a single-line comment symbol, I think. Removing the 'extra' slash clears the syntax highlights but this then makes the source address invalid.

Encountered the same with a another submodule: 

source  = "terraform-aws-modules/cloudwatch/aws//modules/metric-alarm"

Downgraded my plugin to the previous version for now (v2.31.0).

**Comment by dbanck** [comment]
Thanks for the report! I'm able to reproduce the problem.

I believe it has the same root cause as https://github.com/hashicorp/terraform-ls/issues/1771: we're using the top-level schema of the registry module instead of the appropriate submodule one.

I'm looking into a fix

**Comment by dbanck** [comment]
We have just released a new version of vscode-terraform: [2.32.1](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.32.1) that fixes the reported issue. The update should show up automatically in VS Code.

Thanks again for the report @jimmyzzxhlh
